File: Sources/COCHelperCore/CoAPIConfig.swift
import Foundation

/// Connection settings for the official Clash of Clans API.
public struct CoAPIConfig: Sendable, Equatable {
    public var scheme: String
    public var host: String
    public var apiVersion: String
    public var requestTimeout: TimeInterval
    public var maxRetryCount: Int
    public var baseRetryDelay: TimeInterval
    public var maxRetryDelay: TimeInterval

    public init(
        scheme: String = "https",
        host: String = "api.clashofclans.com",
        apiVersion: String = "v1",
        requestTimeout: TimeInterval = 20,
        maxRetryCount: Int = 2,
        baseRetryDelay: TimeInterval = 0.5,
        maxRetryDelay: TimeInterval = 8
    ) {
        self.scheme = scheme
        self.host = host
        self.apiVersion = apiVersion
        self.requestTimeout = requestTimeout
        self.maxRetryCount = maxRetryCount
        self.baseRetryDelay = baseRetryDelay
        self.maxRetryDelay = maxRetryDelay
    }
}

File: packages/domain/src/official/co-api-config.ts
/** 官方 Clash of Clans API 连接配置（对齐 CoAPIConfig.swift）。 */
export type CoAPIConfig = {
  readonly scheme: string;
  readonly host: string;
  readonly apiVersion: string;
  readonly requestTimeoutMs: number;
  readonly maxRetryCount: number;
  readonly baseRetryDelayMs: number;
  readonly maxRetryDelayMs: number;
};

export const DEFAULT_CO_API_CONFIG: CoAPIConfig = {
  scheme: 'https',
  host: 'api.clashofclans.com',
  apiVersion: 'v1',
  requestTimeoutMs: 20_000,
  maxRetryCount: 2,
  baseRetryDelayMs: 500,
  maxRetryDelayMs: 8_000,
};
